php/tests/memory_leak_test.php

```diff
@@ -126,6 +126,18 @@
 $from->setSeconds(1);
 assert(1, $from->getSeconds());
 
+$timestamp = new \Google\Protobuf\Timestamp();
+
+date_default_timezone_set('UTC');
+$from = new DateTime('2011-01-01T15:03:01.012345UTC');
+$timestamp->fromDateTime($from);
+assert($from->format('U') == $timestamp->getSeconds());
+assert(0 == $timestamp->getNanos());
+
+$to = $timestamp->toDateTime();
```

Comment:
I would recommend you add a ``toDateTimeImmutable`` method as well.

Reply:
We can do it in another PR if we decide to add new API. Could you help submit an issue for that?

Reply:
Sorry, I don't have the time for that. I've plenty of things to work on for https://github.com/xdebug/xdebug for PHP 7.2 support.

Reply:
Added #4050 to track it. We will provide support for DateTimeImmutable in next release.
How do u think about adding a boolean arg (is_immutable) to toDateTime instead of providing toDateTimeImmutable? As I see, fromDatetime already takes both DateTime and DateTimeImmutable, it's better to let fromDateTime provide both type of instance.